dace/sdfg/utils.py

```diff
@@ -1046,3 +1046,82 @@ def unique_node_repr(graph: Union[SDFGState, ScopeSubgraphView],
     state = graph if isinstance(graph, SDFGState) else graph._graph
     return str(sdfg.sdfg_id) + "_" + str(sdfg.node_id(state)) + "_" + str(
         state.node_id(node))
+
+
+def update_path_subsets(
+    state: SDFGState,
+    inner_edge_info: Union[MultiConnectorEdge, nd.AccessNode],
+    new_subset: sbs.Subset,
+):
+    """
+    Will take the memlet path defined by :param inner_edge:, and recreate it with the specified new_subset
+    :param inner_edge_info: This is either the inner edge defining the memlet path that
+        should be modified, or an AccessNode with exactly one attached ingoing or outgoing
+        memlet path 
+    :param new_subset: The new subset set on the innermost edge
+    """
+    if isinstance(inner_edge_info, nd.AccessNode):
+        some_edge = list(state.all_edges(inner_edge_info))
+        if len(some_edge) != 1:
+            raise ValueError(
+                "You may not specify an AccessNode in the update_access_list or in "
+                "update_hbm_access_list if it does not have exactly one attached memlet path"
+            )
+        some_edge = some_edge[0]
+        if some_edge.dst == inner_edge_info:
+            inner_edge_info = state.memlet_path(some_edge)[0]
+        else:
+            inner_edge_info = state.memlet_path(some_edge)[-1]
+    mem: mm.Memlet = inner_edge_info.data
+    mem.subset = new_subset
+
+    path = state.memlet_path(inner_edge_info)
+    src_conn = path[0].src_conn
+    dst_conn = path[-1].dst_conn
+    path_nodes = []
+    for edge in path:
+        path_nodes.append(edge.src)
+        state.remove_edge_and_connectors(edge)
+    path_nodes.append(path[-1].dst)
+
+    if src_conn is not None:
+        path[0].src.add_out_connector(src_conn)
+    if dst_conn is not None:
+        path[-1].dst.add_in_connector(dst_conn)
+
+    state.add_memlet_path(*path_nodes,
+                          memlet=mem,
+                          src_conn=src_conn,
+                          dst_conn=dst_conn)
+
+
+def update_array_shape(
+    sdfg: SDFG,
+    array_name: str,
+    new_shape: Iterable,
+    strides=None,
+    offset=None,
+    total_size=None,
+):
```

Comment:
Doesn't work in general since there are other properties (strides, offset, total_size) that depend on the shape, so just directly modifying the shape leads to errors. I'm reinserting to use the default initialization of those values provided by the add_array function

Reply:
I see, maybe we should update the setter of the data container to handle this? Seems like a problem with `array`

Reply:
In principle yes, but I'm not sure if it is clean to just update strides, offset and so on if one overwrites shape. (I guess) in principle they are connected but they may also be set to something else as the default initialization right?

Reply:
Let's bring this up in the meeting

Reply:
Conclusion from the meeting: Make this a method of Array